addons/web/static/src/js/owl_compatibility.js

```diff
@@ -0,0 +1,238 @@
+odoo.define('web.OwlCompatibility', function () {
+    "use strict";
+
+    /**
+     * This file defines the necessary tools for the transition phase where Odoo
+     * legacy widgets and Owl components will coexist. There are two possible
+     * scenarios:
+     *  1) An Owl component has to instantiate legacy widgets
+     *  2) A legacy widget has to instantiate Owl components
+     */
+
+    const { Component, tags } = owl;
+
+    /**
+     * Case 1) An Owl component has to instantiate legacy widgets
+     * ----------------------------------------------------------
+     *
+     * The ComponentAdapter is an Owl component meant to be used as universal
+     * adapter for Owl components that embed Odoo legacy widgets (or dynamically
+     * both Owl components and Odoo legacy widgets), e.g.:
+     *
+     *                           Owl Component
+     *                                 |
+     *                         ComponentAdapter (Owl component)
+     *                                 |
+     *                       Legacy Widget(s) (or Owl component(s))
+     *
+     *
+     * The adapter takes the component/widget class as 'Component' prop, and the
+     * arguments (except first arg 'parent') to initialize it as props.
+     * For instance:
+     *     <ComponentAdapter Component="LegacyWidget" params="params"/>
```

Comment:
I propose a hardcoded `args` props, that would be given to the widget constructor, so we do not need to subclass the adapter in that case